app/streaming/video/ffmpeg-renderers/d3d11va.cpp

```diff
@@ -150,11 +174,130 @@ D3D11VARenderer::~D3D11VARenderer()
     else {
         SAFE_COM_RELEASE(m_Device);
         SAFE_COM_RELEASE(m_DeviceContext);
+        SAFE_COM_RELEASE(m_VideoDevice);
+        SAFE_COM_RELEASE(m_VideoContext);
     }
 
     SAFE_COM_RELEASE(m_Factory);
 }
 
+/**
+ * \brief Set Monitor HDR MetaData information
+ *
+ * Get the Monitor HDT MetaData via LimeLight library
+ *
+ * \param PSS_HDR_METADATA* HDRMetaData The varaible to set the metadata information
+ * \return bool Return True is succeed
+ */
+void D3D11VARenderer::setHDRStream(){
+    DXGI_HDR_METADATA_HDR10 streamHDRMetaData;
+    // Prepare HDR Meta Data for Stream content
+    SS_HDR_METADATA hdrMetadata;
+    // if (m_VideoProcessor.p && m_IsHDRenabled && LiGetHdrMetadata(&hdrMetadata)) {
+    if (m_VideoProcessor.p && LiGetHdrMetadata(&hdrMetadata)) {
+
+        // Magic constants to convert to fixed point.
+        // https://docs.microsoft.com/en-us/windows/win32/api/dxgi1_5/ns-dxgi1_5-dxgi_hdr_metadata_hdr10
+        static constexpr int kMinLuminanceFixedPoint = 10000;
+
+        streamHDRMetaData.RedPrimary[0] = hdrMetadata.displayPrimaries[0].x;
+        streamHDRMetaData.RedPrimary[1] = hdrMetadata.displayPrimaries[0].y;
+        streamHDRMetaData.GreenPrimary[0] = hdrMetadata.displayPrimaries[1].x;
+        streamHDRMetaData.GreenPrimary[1] = hdrMetadata.displayPrimaries[1].y;
+        streamHDRMetaData.BluePrimary[0] = hdrMetadata.displayPrimaries[2].x;
+        streamHDRMetaData.BluePrimary[1] = hdrMetadata.displayPrimaries[2].y;
+        streamHDRMetaData.WhitePoint[0] = hdrMetadata.whitePoint.x;
+        streamHDRMetaData.WhitePoint[1] = hdrMetadata.whitePoint.y;
+        streamHDRMetaData.MaxMasteringLuminance = hdrMetadata.maxDisplayLuminance * kMinLuminanceFixedPoint;
+        streamHDRMetaData.MinMasteringLuminance = hdrMetadata.minDisplayLuminance;
+
+        streamHDRMetaData.MaxContentLightLevel = hdrMetadata.maxContentLightLevel;
+        streamHDRMetaData.MaxFrameAverageLightLevel = hdrMetadata.maxFrameAverageLightLevel;
+        if(streamHDRMetaData.MaxContentLightLevel == 0){
+            streamHDRMetaData.MaxContentLightLevel = streamHDRMetaData.MaxFrameAverageLightLevel;
+        }
```

Comment:
Ok, having those 2 values to zero doesn't seem to impact the current visual rendering.
I can keep those lines of code in case it is used in the future.

Reply:
I finally set those 2 lines at 0 by default as MaxCLL and MaxFALL only make sense for a pre-defined content (calculated values based on content), cannot be evaluated on the fly.